src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -1347,12 +1347,14 @@ private static bool SatisfiesConstructorConstraint(TypeSymbol typeArgument)
             switch (typeArgument.TypeKind)
             {
                 case TypeKind.Struct:
+                    return HasPublicParameterlessConstructor((NamedTypeSymbol)typeArgument, synthesizedIfMissing: true);
```

Comment:
Just curious, do you know why the class case has a type argument check, but the struct case doesn't need one?

Reply:
The `typeArgument` is the `struct` type and cannot be abstract.